geode-wan/src/main/java/org/apache/geode/cache/wan/internal/GatewaySenderFactoryImpl.java

```diff
@@ -256,7 +257,13 @@ public GatewaySenderFactory setEnforceThreadsConnectSameReceiver(
 
   static void validate(final @NotNull InternalCache cache,
       final @NotNull GatewaySenderAttributesImpl attributes) {
-    final int myDSId = cache.getDistributionManager().getDistributedSystemId();
+    int myDSId;
+    if (cache instanceof CacheCreation) {
```

Comment:
Maybe `CacheCreation` should be fixed to behave like any other `InternalCache` rather than making consumer know subtile differences in behavior.

Reply:
You mean to implement in `CacheCreation` the `getDistributionManager()` method as follows?
```
  @Override
  public DistributionManager getDistributionManager() {
    return InternalDistributedSystem.getAnyInstance().getDistributionManager()
  }

```